crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -500,14 +529,22 @@ impl PendingTransactionsReceiver {
         pending_txs
     }
 }
+/// Represents the kind of pending transaction data that can be retrieved.
+///
+/// This enum differentiates between two kinds of pending transaction data:
+/// - Just the transaction hashes.
+/// - Full transaction details.
+enum PendingTransactionKind<T: PoolTransaction> {
+    Hashes(PendingTransactionsReceiver),
+    FullTransaction(FullTransactionsReceiver<T>),
+}
 
 #[derive(Clone, Debug)]
 enum FilterKind {
```

Comment:
here we need to make the FilterKind generic over T so we can use 

`PendingTransactionsReceiver<T>`

this also needs to be updated where it's used with `Poll::Transaction`

Reply:
> FilterKind

lot of others code will be breaked if we make FilterKind generic? I was going to make FilterKind as generic but saw lots of error. Do you suggest to use generic for FilterKind  ?

Reply:
yeah if i change FilterKind to generic this would break all others type that depend in FilterKind :)

what do you think of this?

Reply:
since this type is private, I think we can easily do this, even if more verbose

Reply:
> since this type is private, I think we can easily do this, even if more verbose

hello Mat, hope your days going good. Made some changes in filter.rs, is this is the approach that i need to follow ? 

i think i'm in the last step for finishing but now i had a problem which is 


<Pool as TransactionPool>::Transaction does not impl the Clone trait, but it's required by the FilterKind enum definition for its type parameter A.

Do you have any suggestion for this please? thank you for all your help doing. I really appreciate and i really love working on this 
Much love and peace Mat !